back-end/helpers/intent_analyzer.js

```diff
@@ -0,0 +1,56 @@
+export default class IntentAnalyzer{
+    /* What it should result in:
+    [
+        {
+          question: "What is something that Waluigi and Wario often say?",
+          children: {
+            "The brothers Waluigi and Wario have their catchphrase 'waaa'.": 100,
+          }
+        },
+        {
+          question: "Where should I go for a good time and great learning experience?",
+          children: {
+            "You should go to the University of Toronto.": 10,
+            "You should go to the Technology Leadership Initiative.": 90
+          }
+        }, 
+        {
+          question: "Hey BestFlow, what time is it?",
+          children: {
+            "It's time for you to get a watch, would you like to take a look at our product?": 10,
+            "Showtime showtime what I'm John Laurens in the place to be.": 20,
+            "Summertime! School's out, scream and shout.": 70
+          }
+        }
+    ]
+      */
+
+    static async formatIntents(intents) { // Formats the mongoose Model into desired array for IntentLister to use
```

Comment:
I tried researching this, but I still don't quite understand what is the best practice: when is it good to use static async and when is it not? A lot of the javascript coding I've learned was by looking at the other files and a lot of them use static async so- :')

Reply:
We use `static` when all instances of intent analyzer need to share a piece of code (i.e. incrementing some universal variable). To my understanding, we only need to use `async` if we expect parts of the code to be asynchronous, and there is nothing in either of the methods here that looks like it (usually we will see an `await` keyword if something is async).

So definitely remove `async`, I think you can remove `static` as well because we should only have 1 instance of the intent analyzer so it doesn't matter.